node/cli/src/command.rs

```diff
@@ -704,58 +707,93 @@ pub fn run() -> Result<()> {
 				};
 
 				let extension = chain_spec::Extensions::try_get(&*config.chain_spec);
-				let para_id = extension.map(|e| e.para_id);
-				let id = ParaId::from(cli.run.parachain_id.clone().or(para_id).unwrap_or(1000));
 
 				let rpc_config = cli.run.new_rpc_config();
 
-				// If dev service was requested, start up manual or instant seal.
-				// Otherwise continue with the normal parachain node.
-				// Dev service can be requested in two ways.
-				// 1. by providing the --dev-service flag to the CLI
-				// 2. by specifying "dev-service" in the chain spec's "relay-chain" field.
-				// NOTE: the --dev flag triggers the dev service by way of number 2
-				let relay_chain_id = extension.map(|e| e.relay_chain.as_str());
-				let dev_service = cli.run.dev_service
-					|| config.chain_spec.is_dev()
-					|| relay_chain_id == Some("dev-service");
-
-				if dev_service {
+				#[cfg(feature = "lazy-loading")]
+				if cli.run.fork_chain_from_rpc.is_some() {
 					// When running the dev service, just use Alice's author inherent
 					//TODO maybe make the --alice etc flags work here, and consider bringing back
 					// the author-id flag. For now, this will work.
 					let author_id = Some(chain_spec::get_from_seed::<nimbus_primitives::NimbusId>(
 						"Alice",
 					));
 
-					return match &config.chain_spec {
-						#[cfg(feature = "moonriver-native")]
-						spec if spec.is_moonriver() => moonbeam_service::new_dev::<
-							moonbeam_service::moonriver_runtime::RuntimeApi,
-							moonbeam_service::MoonriverCustomizations,
-							sc_network::NetworkWorker<_, _>,
-						>(config, author_id, cli.run.sealing, rpc_config, hwbench)
-						.await
-						.map_err(Into::into),
-						#[cfg(feature = "moonbeam-native")]
-						spec if spec.is_moonbeam() => moonbeam_service::new_dev::<
-							moonbeam_service::moonbeam_runtime::RuntimeApi,
-							moonbeam_service::MoonbeamCustomizations,
-							sc_network::NetworkWorker<_, _>,
-						>(config, author_id, cli.run.sealing, rpc_config, hwbench)
-						.await
-						.map_err(Into::into),
-						#[cfg(feature = "moonbase-native")]
-						_ => moonbeam_service::new_dev::<
-							moonbeam_service::moonbase_runtime::RuntimeApi,
-							moonbeam_service::MoonbaseCustomizations,
-							sc_network::NetworkWorker<_, _>,
-						>(config, author_id, cli.run.sealing, rpc_config, hwbench)
-						.await
-						.map_err(Into::into),
-						#[cfg(not(feature = "moonbase-native"))]
-						_ => panic!("invalid chain spec"),
+					let lazy_loading_config = moonbeam_cli_opt::LazyLoadingConfig {
+						state_rpc: cli.run.fork_chain_from_rpc.expect("a valid RPC endpoint"),
```

Comment:
If you pass a wrong RPC or even a malformed url, it never reaches this error, idk why it crashes somewhere else

Reply:
can we track this in a separate issue @gonzamontiel ?